src/main/java/com/dougnoel/sentinel/files/CsvFile.java

```diff
@@ -0,0 +1,334 @@
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile() throws FileNotFoundException {
+        super();
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create default CSV file from specified path.
+     * @param pathToFile Path path to the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file with the given file format from the CSV file at the specified path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the most recently downloaded path.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(int numberOfHeaderRows) throws FileNotFoundException {
+        super();
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows and CSVFormat from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
1. this is an auto-generated method created by IntelliJ. It's recommended that any non-static class implements an equals() method so it returns accurately based on the fields and such on the object.
2. If I understand java correctly, the first comparison (line 96) does a check based on the object in the java heap. It will return true if the compiler's reference in the heap to both objects is the same. Not sure what you mean by loading issues. It's strictly checking whether the object passed exists in the same location in memory as the current object. if so, that means they're the exact same object.
3. the last line returns true or false, depending on whether the object passed in has values in those fields that match the current instance's values. basically, it will only reach that line if it's determined that the object passed in is the same type (CsvFile) but not the same memory location, and if those values match the current instance's, then we can say the two objects are equal.
hope that helps.

Reply:
Oh, I had no idea it was auto-generated. Lets keep it the way it is then. I just read up on java == for objects, I think you're right about that. That explains the last line too, I thought we were just comparing counts instead of the whole row. Thanks!